File: app/src/main/java/xyz/gsora/toot/RealmBuilder.java
package xyz.gsora.toot;

import io.realm.Realm;
import io.realm.RealmConfiguration;

import static xyz.gsora.toot.Timeline.Companion;

/**
 * Created by gsora on 5/5/17.
 * <p>
 * Build a Realm configuration based on the destination timeline content
 */
public class RealmBuilder {

    public static Realm getRealmForTimelineContent(Timeline.TimelineContent timelineContent) {
        RealmConfiguration config = null;

        switch (timelineContent) {
            case TIMELINE_MAIN:
                config = new RealmConfiguration.Builder().name(Companion.getTIMELINE_MAIN()).build();
                break;
            case TIMELINE_LOCAL:
                config = new RealmConfiguration.Builder().name(Companion.getTIMELINE_LOCAL()).build();
                break;
            case TIMELINE_FEDERATED:
                config = new RealmConfiguration.Builder().name(Companion.getTIMELINE_FEDERATED()).build();
                break;
            case FAVORITES:
                config = new RealmConfiguration.Builder().name(Companion.getFAVORITES()).build();
                break;
            case NOTIFICATIONS:
                config = new RealmConfiguration.Builder().name(Companion.getNOTIFICATIONS()).build();
                break;
        }

        return Realm.getInstance(config);
    }
}

File: app/src/main/java/xyz/gsora/toot/RealmBuilder.kt
package xyz.gsora.toot

import io.realm.Realm
import io.realm.RealmConfiguration

import xyz.gsora.toot.Timeline.Companion

/**
 * Created by gsora on 5/5/17.
 *
 *
 * Build a Realm configuration based on the destination timeline content
 */
object RealmBuilder {

    fun getRealmForTimelineContent(timelineContent: Timeline.TimelineContent): Realm {
        var config: RealmConfiguration? = null

        when (timelineContent) {
            Timeline.TimelineContent.TIMELINE_MAIN -> config = RealmConfiguration.Builder().name(Companion.TIMELINE_MAIN).build()
            Timeline.TimelineContent.TIMELINE_LOCAL -> config = RealmConfiguration.Builder().name(Companion.TIMELINE_LOCAL).build()
            Timeline.TimelineContent.TIMELINE_FEDERATED -> config = RealmConfiguration.Builder().name(Companion.TIMELINE_FEDERATED).build()
            Timeline.TimelineContent.FAVORITES -> config = RealmConfiguration.Builder().name(Companion.FAVORITES).build()
            Timeline.TimelineContent.NOTIFICATIONS -> config = RealmConfiguration.Builder().name(Companion.NOTIFICATIONS).build()
            Timeline.TimelineContent.List -> config = RealmConfiguration.Builder().name(Companion.List).build()
            Timeline.TimelineContent.HASHCODE -> config = RealmConfiguration.Builder().name(Companion.Hashcode).build()
        }

        return Realm.getInstance(config!!)
    }
}
